dotCMS/src/main/java/com/dotmarketing/fixtask/tasks/FixTask00095DeleteOrphanRelationships.java

```diff
@@ -0,0 +1,221 @@
+        
+        FixAssetsProcessStatus.setTotal(total);
+        return total > 0 ? true : false;
+    }
+
+    
+    private void startCleanup() throws DotDataException {
+        try {
+            
+            cleanUpOrphanRelationships();
+            // Set the number of records that were fixed
+            FixAssetsProcessStatus.setErrorsFixed(modifiedData.size());
+        } catch (SQLException | DotHibernateException e1) {
+            Logger.error(this, "There was an unexpected problem with cleaning up relationship in Database.", e1);
+            modifiedData.clear();
+        }
+    }
+    
+    private void saveFixAuditRecord(final int total) throws DotHibernateException {
+        final FixAudit auditObj = new FixAudit();
+        auditObj.setTableName("relationship");
+        auditObj.setDatetime(new Date());
+        auditObj.setRecordsAltered(total);
+        auditObj.setAction("task 95: DeleteOrphanRelationships");
+        HibernateUtil.save(auditObj);
+    }
+    
+    private void cleanUpOrphanRelationships() throws SQLException, DotDataException{
+        
+        
+        String query1 = "DELETE FROM relationship WHERE parent_structure_inode = ?";
+        String query2 = "DELETE FROM relationship WHERE child_structure_inode = ?";
+        String query3 = String.format("DELETE FROM inode WHERE NOT EXISTS (SELECT * FROM relationship " + 
+                "WHERE relationship.inode = inode.inode) and inode.type like '%s' ",Inode.Type.RELATIONSHIP.getTableName());
+        
+        List<Relationship> relsToClearInCache = new ArrayList<>();
+        
+        final DotConnect dc = new DotConnect();
+        dc.setSQL(VERIFICATION_QUERY);
+        List<Map<String,Object>> results = dc.loadObjectResults();
+        
+        List<String> relInodes = ((Map<String, String>) results).values()
+                .stream().map(Object::toString).collect(Collectors.toList());
+
+        relInodes.forEach(item->{
+            try {
+                Relationship rel = APILocator.getRelationshipAPI().byInode(item);
+                relsToClearInCache.add(rel);    
+                
+                dc.setSQL(query1);
+                dc.addParam(item);        
+                dc.loadResult();
+                
+                dc.setSQL(query2);
+                dc.addParam(item);        
+                dc.loadResult();
+            } catch (DotDataException e){
+                Logger.error(this, "There was an unexpected problem with cleaning up relationship in Database.", e);
+                modifiedData.clear();
+            }
+        });
+        
+        dc.executeStatement(query3);
+        flushRelationshipCacheRegion(relsToClearInCache);
+    }
+    
+    private void flushRelationshipCacheRegion(List<Relationship> inodesToClearInCache) throws DotHibernateException {
+        HibernateUtil.addCommitListener(new FlushCacheRunnable() {
+            public void run () {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'rel' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)